controllers/activemqartemisservice_test.go

```diff
@@ -0,0 +1,484 @@
+				// minikube> kubectl port-forward svc/debug 5005:5005 --namespace test
+				By("setup debug")
+				debugService = &corev1.Service{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Service",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "debug",
+						Namespace: defaultNamespace,
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeNodePort,
+						Selector: map[string]string{
+							"ActiveMQArtemis": crd.Name,
+						},
+						Ports: []corev1.ServicePort{
+							{
+								Port: 5005,
+							},
+						},
+					},
+				}
+				Expect(k8sClient.Create(ctx, debugService)).Should(Succeed())
+			}
+
+			serviceKey := types.NamespacedName{Name: crd.Name, Namespace: crd.Namespace}
+			createdCrd := &brokerv1beta1.ActiveMQArtemisService{}
+
+			By("Checking ready cr status")
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, serviceKey, createdCrd)).Should(Succeed())
+
+				if verbose {
+					fmt.Printf("Service STATUS: %v\n\n", createdCrd.Status.Conditions)
+				}
+				g.Expect(meta.IsStatusConditionTrue(createdCrd.Status.Conditions, brokerv1beta1.ReadyConditionType)).Should(BeTrue())
+
+			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+			By("checking broker cr status insync - should be in the service if important to user")
+			brokerKey := types.NamespacedName{Name: crd.Name, Namespace: crd.Namespace}
+			brokerCrd := &brokerv1beta1.ActiveMQArtemis{}
+
+			var appPropsRv = ""
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, brokerKey, brokerCrd)).Should(Succeed())
+
+				if verbose {
+					fmt.Printf("Broker STATUS: %v\n\n", brokerCrd.Status)
+				}
+
+				condition := meta.FindStatusCondition(brokerCrd.Status.Conditions, brokerv1beta1.ConfigAppliedConditionType)
+				g.Expect(condition).NotTo(BeNil())
+
+				for _, externalConfig := range brokerCrd.Status.ExternalConfigs {
+					if externalConfig.Name == AppPropertiesSecretName(brokerCrd.Name) {
+						appPropsRv = externalConfig.ResourceVersion
+					}
+				}
+				g.Expect(appPropsRv).ShouldNot(BeEmpty())
+
+			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+			By("deploying a matching app")
+			appName := "first-app" // a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')",
+			app := brokerv1beta1.ActiveMQArtemisApp{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ActiveMQArtemisApp",
+					APIVersion: brokerv1beta1.GroupVersion.Identifier(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      appName,
+					Namespace: defaultNamespace,
+				},
+				Spec: brokerv1beta1.ActiveMQArtemisAppSpec{
+
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"forWorkQueue": "true",
```

Comment:
I think it would be better if `forWorkQueue` would be in small caps `forworkqueue` in order to match the style of the casing in the yaml:
I''m also wondering the same for `matchLabels` that is using camel case.

```yaml
kubectl apply -f - <<EOF
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemisApp
metadata:
  name: first-app
  namespace: service-app-project
spec:
  selector:
    matchLabels:
      forWorkQueue: "true"
  auth:
    - mtls
  capabilities:
    - producerandconsumerof:
      - queuename: "APP.JOBS"
EOF
```